java/org/contikios/cooja/Cooja.java

```diff
@@ -1470,31 +1486,23 @@ Element extractSimulationConfig() {
       if (startedPlugin.getCooja() != null) {
         JInternalFrame pluginFrame = startedPlugin.getCooja();
 
-        var pluginSubElement = new Element("width");
-        pluginSubElement.setText(String.valueOf(pluginFrame.getSize().width));
-        pluginElement.addContent(pluginSubElement);
-
-        pluginSubElement = new Element("z");
-        pluginSubElement.setText(String.valueOf(getDesktopPane().getComponentZOrder(pluginFrame)));
-        pluginElement.addContent(pluginSubElement);
-
-        pluginSubElement = new Element("height");
-        pluginSubElement.setText(String.valueOf(pluginFrame.getSize().height));
-        pluginElement.addContent(pluginSubElement);
-
-        pluginSubElement = new Element("location_x");
-        pluginSubElement.setText(String.valueOf(pluginFrame.getLocation().x));
-        pluginElement.addContent(pluginSubElement);
+        var pluginSubElement = new Element("bounds");
+        var bounds = pluginFrame.getBounds();
+        pluginSubElement.setAttribute("x", Integer.toString(bounds.x));
```

Comment:
I think String.valueOf() does the same, and if bounds.x changes type the code will stil work.

Reply:
`String.valueOf(int)` calls `Integer.toString(int)`, so it does the same. As the parsing expects this to be an integer value, I wanted to enforce this in the generation, but I do not have anything against `String.valueOf()` and can update this.

Reply:
Well, if something changes type, there's less code to update with `String.valueOf()` (and the same method works for the true boolean in line 1502, and the doubles in Position.java).

Reply:
Updated.